openlibrary/plugins/upstream/models.py

```diff
@@ -489,6 +490,48 @@ def is_fake_record(self):
         """
         return "/ia:" in self.key
 
+    def as_json_ld(self):
+        def get_identifier(label):
+            identifier = list(values[0].value for name, values in self.get_identifiers().multi_items() if values[0].label == label)
+            identifier = identifier[0] if identifier else None
+            return identifier
+
+        def get_url(label):
+            url = list(values[0].url for name, values in self.get_identifiers().multi_items() if values[0].label == label)
+            url = url[0] if url else None
+            return url
+    
+        return json.dumps({
+            "@context": "http://schema.org/",
+            "@type": "Book",
+            "@id": web.ctx.homedomain + self.key,
+            "workExample": [
+                {
+                    "@type": "Book",
+                    "@id": web.ctx.homedomain + self.key,
+                    "isbn": self.get_isbn13(),
+                    "image": [
+                         self.get_cover_url("L")
+                    ],
+                    "bookFormat": "http://schema.org/"+ self.physical_format.replace("[", "").replace("]", "") if self.physical_format else None,
+                    "inLanguage": (", ".join(lang.code for lang in self.languages)) or None,
+                    "datePublished": self.publish_date or None,
+                    "sameAs":[
+                        get_url("OCLC/WorldCat"),
+                        get_url("Library Thing"),
+                        get_url("Goodreads")
+                    ],
+                    "identifier":[
+                        {
+                        "@type": "PropertyValue",
+                        "propertyID": "OCLC_NUMBER",
+                        "value": get_identifier("OCLC/WorldCat")
+                        }
+                    ]
```

Comment:
Also place this in an if block so we don't include it if the OCLC is not specified.